server/app/services/question/types/MultiOptionQuestionDefinitionConfig.java

```diff
@@ -0,0 +1,88 @@
+package services.question.types;
+
+import com.google.auto.value.AutoValue;
+import com.google.common.collect.ImmutableList;
+import java.time.Instant;
+import java.util.Optional;
+import java.util.OptionalLong;
+import services.LocalizedStrings;
+import services.question.QuestionOption;
+import services.question.types.MultiOptionQuestionDefinition.MultiOptionQuestionType;
+import services.question.types.MultiOptionQuestionDefinition.MultiOptionValidationPredicates;
+
+@AutoValue
+public abstract class MultiOptionQuestionDefinitionConfig {
+  abstract OptionalLong id();
+
+  abstract String name();
+
+  abstract Optional<Long> enumeratorId();
+
+  abstract String description();
+
+  abstract LocalizedStrings questionText();
+
+  abstract LocalizedStrings questionHelpText();
+
+  abstract ImmutableList<QuestionOption> questionOptions();
+
+  abstract Optional<MultiOptionValidationPredicates> validationPredicates();
+
+  abstract Optional<Instant> lastModifiedTime();
+
+  abstract MultiOptionQuestionType multiOptionQuestionType();
+
+  public static RequiredMultiOptionQuestionType builder() {
+    return new AutoValue_MultiOptionQuestionDefinitionConfig.Builder();
+  }
+
+  static final MultiOptionValidationPredicates SINGLE_SELECT_PREDICATE =
+      MultiOptionValidationPredicates.create(1, 1);
+
+  public interface RequiredMultiOptionQuestionType {
+    Builder setMultiOptionQuestionType(MultiOptionQuestionType multiOptionQuestionType);
+  }
+
```

Comment:
The `RequiredMultiOptionQuestionType` interface simply forces the creator of a config to specify a question type, like:

```java
    MultiOptionQuestionDefinitionConfig config =
        MultiOptionQuestionDefinitionConfig.builder()
            .setMultiOptionQuestionType(MultiOptionQuestionType.CHECKBOX)
            .setName("name")
            .setEnumeratorId(Optional.empty())
            .setDescription("description")
            .setQuestionText(questionText)
            .setQuestionHelpText(questionHelpText)
            .setQuestionOptions(options)
            .setValidationPredicates(validationPredicates)
            .build();
```

Reply:
Makes sense, thanks! Could you add a javadoc?

Reply:
Done.